src/monevade.cpp

```diff
@@ -0,0 +1,43 @@
+#include "monevade.h"
+
+#include <algorithm>
+
+#include "creature.h"
+#include "game.h"
+#include "messages.h"
+#include "monster.h"
+#include "output.h"
+#include "player.h"
+#include "projectile.h"
+#include "rng.h"
+#include "translations.h"
+
+const efftype_id effect_stunned( "stunned" );
+
+void mevade::none( monster &, Creature *, const dealt_projectile_attack * )
+{
+}
+
+void mevade::telestagger( monster &m, Creature *const source,
+                          dealt_projectile_attack const *proj )
+{
+    if( source == nullptr ) {
+        return;
+    }
+    // Can't stagger on ranged attacks.
```

Comment:
What about melee reach attack?

Reply:
Changed logic to more explicitly support, and changed comment.